Implement GPU process side display lists

kkinnunen-apple opened the pull request:
#### b69cd22bb96f0e0ab9487cd14f3aa47fbeef02bc
<pre>
Implement GPU process side display lists
<a href="https://bugs.webkit.org/show_bug.cgi?id=297727">https://bugs.webkit.org/show_bug.cgi?id=297727</a>
<a href="https://rdar.apple.com/158865006">rdar://158865006</a>

Reviewed by Simon Fraser.

Cache DisplayList data in GPUP and refer to them via
RemoteDisplayListIdentifier. Caching is done by constructing a
RemoteDisplayListRecorder IPC receiver that forwards the draw calls
to DisplayList::RecorderImpl to construct the DisplayList::DisplayList
instance.

Remove DecomposedGlyphs, it was overlapping and overly narrow
functionality.

Before, for a specific glyph run GlyphDisplayListCache would
drawGlyphs() to a DisplayList::RecorderImpl in DecomposeAndRetain mode.
This would decompose the single DrawGlyph call into Translate, Scale,
ConcatenateCTM, DrawDecomposedGlyphs, ... operations.
DrawDecomposedGlyphs would contain the DecomposedGlyphs structure, which
would be cached to GPUP upon first playback of the display list. Further
playbacks would again issue all commands of the display list, but reuse
the cached DecomposedGlyphs data.

After, we cache the whole display list. This way drawing the display
list is just one command.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/graphics/BifurcatedGraphicsContext.cpp:
(WebCore::BifurcatedGraphicsContext::drawDecomposedGlyphs): Deleted.
* Source/WebCore/platform/graphics/BifurcatedGraphicsContext.h:
* Source/WebCore/platform/graphics/DecomposedGlyphs.cpp: Removed.
* Source/WebCore/platform/graphics/DecomposedGlyphs.h: Removed.
* Source/WebCore/platform/graphics/FontCascade.cpp:
(WebCore::FontCascade::displayListForTextRun const):
* Source/WebCore/platform/graphics/GraphicsContext.cpp:
(WebCore::GraphicsContext::drawGlyphsImmediate):
(WebCore::GraphicsContext::drawDecomposedGlyphs): Deleted.
* Source/WebCore/platform/graphics/GraphicsContext.h:
* Source/WebCore/platform/graphics/NullGraphicsContext.h:
* Source/WebCore/platform/graphics/RenderingResource.h:
(WebCore::RenderingResource::isGradient const):
(WebCore::RenderingResource::isDecomposedGlyphs const): Deleted.
* Source/WebCore/platform/graphics/cairo/CairoOperationRecorder.cpp:
(WebCore::Cairo::OperationRecorder::drawDecomposedGlyphs): Deleted.
* Source/WebCore/platform/graphics/cairo/CairoOperationRecorder.h:
* Source/WebCore/platform/graphics/cairo/GraphicsContextCairo.cpp:
* Source/WebCore/platform/graphics/coretext/DrawGlyphsRecorder.cpp:
(WebCore::DrawGlyphsRecorder::DrawGlyphsRecorder):
(WebCore::DrawGlyphsRecorder::recordDrawGlyphs):
* Source/WebCore/platform/graphics/coretext/DrawGlyphsRecorder.h:
* Source/WebCore/platform/graphics/displaylists/DisplayList.cpp:
(WebCore::DisplayList::DisplayList::~DisplayList):
* Source/WebCore/platform/graphics/displaylists/DisplayList.h:
* Source/WebCore/platform/graphics/displaylists/DisplayListItem.h:
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.cpp:
(WebCore::DisplayList::DrawDecomposedGlyphs::apply const): Deleted.
(WebCore::DisplayList::DrawDecomposedGlyphs::dump const): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.h:
(WebCore::DisplayList::DrawDecomposedGlyphs::DrawDecomposedGlyphs): Deleted.
(WebCore::DisplayList::DrawDecomposedGlyphs::font const): Deleted.
(WebCore::DisplayList::DrawDecomposedGlyphs::decomposedGlyphs const): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorder.cpp:
(WebCore::DisplayList::Recorder::appendDisplayList):
(WebCore::DisplayList::Recorder::decomposeDrawGlyphsIfNeeded):
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorder.h:
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorderImpl.cpp:
(WebCore::DisplayList::RecorderImpl::drawDecomposedGlyphs): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorderImpl.h:
* Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp:
* Source/WebCore/rendering/GlyphDisplayListCache.cpp:
(WebCore::GlyphDisplayListCache::canShareDisplayList):
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
(WebCore::RenderThemeCocoa::paintAttachmentText):
* Source/WebKit/GPUProcess/graphics/RemoteGraphicsContext.cpp:
(WebKit::RemoteGraphicsContext::RemoteGraphicsContext):
(WebKit::RemoteGraphicsContext::controlFactory):
(WebKit::RemoteGraphicsContext::drawDisplayList):
(WebKit::RemoteGraphicsContext::drawControlPart):
(WebKit::RemoteImageBufferGraphicsContext::create):
(WebKit::RemoteImageBufferGraphicsContext::RemoteImageBufferGraphicsContext):
(WebKit::RemoteImageBufferGraphicsContext::startListeningForIPC):
(WebKit::RemoteImageBufferGraphicsContext::stopListeningForIPC):
(WebKit::RemoteDisplayListRecorder::create):
(WebKit::RemoteDisplayListRecorder::RemoteDisplayListRecorder):
(WebKit::RemoteDisplayListRecorder::startListeningForIPC):
(WebKit::RemoteDisplayListRecorder::stopListeningForIPC):
(WebKit::RemoteGraphicsContext::create): Deleted.
(WebKit::RemoteGraphicsContext::startListeningForIPC): Deleted.
(WebKit::RemoteGraphicsContext::stopListeningForIPC): Deleted.
(WebKit::RemoteGraphicsContext::drawDecomposedGlyphs): Deleted.
* Source/WebKit/GPUProcess/graphics/RemoteGraphicsContext.h:
(WebKit::RemoteGraphicsContext::context):
* Source/WebKit/GPUProcess/graphics/RemoteGraphicsContext.messages.in:
* Source/WebKit/GPUProcess/graphics/RemoteImageBuffer.cpp:
(WebKit::RemoteImageBuffer::RemoteImageBuffer):
* Source/WebKit/GPUProcess/graphics/RemoteImageBuffer.h:
* Source/WebKit/GPUProcess/graphics/RemoteImageBufferSet.cpp:
(WebKit::RemoteImageBufferSet::ensureBufferForDisplay):
* Source/WebKit/GPUProcess/graphics/RemoteImageBufferSet.h:
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.cpp:
(WebKit::RemoteRenderingBackend::createDisplayListRecorder):
(WebKit::RemoteRenderingBackend::sinkDisplayListRecorderIntoDisplayList):
(WebKit::RemoteRenderingBackend::releaseDisplayList):
(WebKit::RemoteRenderingBackend::cacheDecomposedGlyphs): Deleted.
(WebKit::RemoteRenderingBackend::releaseDecomposedGlyphs): Deleted.
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.h:
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.messages.in:
* Source/WebKit/GPUProcess/graphics/RemoteResourceCache.cpp:
(WebKit::RemoteResourceCache::cacheDisplayList):
(WebKit::RemoteResourceCache::cachedDisplayList const):
(WebKit::RemoteResourceCache::releaseDisplayList):
(WebKit::RemoteResourceCache::releaseMemory):
(WebKit::RemoteResourceCache::cacheDecomposedGlyphs): Deleted.
(WebKit::RemoteResourceCache::releaseDecomposedGlyphs): Deleted.
(WebKit::RemoteResourceCache::cachedDecomposedGlyphs const): Deleted.
* Source/WebKit/GPUProcess/graphics/RemoteResourceCache.h:
* Source/WebKit/Scripts/webkit/messages.py:
(atomic_object_identifier):
(serialized_identifiers):
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/graphics/RemoteDisplayListIdentifier.h: Copied from Source/WebKit/WebProcess/GPU/graphics/RemoteGraphicsContextIdentifier.h.
* Source/WebKit/WebProcess/GPU/graphics/RemoteGraphicsContextIdentifier.h:
* Source/WebKit/WebProcess/GPU/graphics/RemoteGraphicsContextProxy.cpp:
(WebKit::RemoteGraphicsContextProxy::RemoteGraphicsContextProxy):
(WebKit::m_renderingBackend):
(WebKit::RemoteGraphicsContextProxy::drawDisplayList):
(WebKit::RemoteGraphicsContextProxy::recordResourceUse):
(WebKit::RemoteGraphicsContextProxy::drawDecomposedGlyphs): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/RemoteGraphicsContextProxy.h:
* Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.cpp:
(WebKit::RemoteRenderingBackendProxy::cacheDisplayList):
(WebKit::RemoteRenderingBackendProxy::releaseDisplayList):
(WebKit::RemoteRenderingBackendProxy::cacheDecomposedGlyphs): Deleted.
(WebKit::RemoteRenderingBackendProxy::releaseDecomposedGlyphs): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.h:
* Source/WebKit/WebProcess/GPU/graphics/RemoteResourceCacheProxy.cpp:
(WebKit::RemoteResourceCacheProxy::recordDisplayListUse):
(WebKit::RemoteResourceCacheProxy::willDestroyDisplayList):
(WebKit::RemoteResourceCacheProxy::releaseMemory):
(WebKit::RemoteResourceCacheProxy::recordDecomposedGlyphsUse): Deleted.
(WebKit::RemoteResourceCacheProxy::willDestroyDecomposedGlyphs): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/RemoteResourceCacheProxy.h:

Canonical link: <a href="https://commits.webkit.org/299747@main">https://commits.webkit.org/299747@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/64c9de0598cc443a55d8edb9791fcb91d508ee75

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/120043 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/39737 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/30388 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/126382 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/72116 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/c1b4d5c4-7ef9-4db6-a3c2-27ae1b04d925) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/121919 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/40433 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/48314 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91167 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/60475 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/2a157d00-d2a8-4c55-a685-5110bc22ed03) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/122995 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32297 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/107638 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/71720 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/6e6c5589-1fc2-469f-88a9-80b5a0ab90e2) 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/119400 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/31332 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/25742 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/70013 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/101776 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25930 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/129293 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/46963 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/35619 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/99778 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/47329 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103825 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/99622 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25298 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45081 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23101 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/43587 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/46825 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/52531 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/46291 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/49640 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/47977 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebKit/GPUProcess/graphics/RemoteGraphicsContext.cpp:
```diff
@@ -767,6 +749,58 @@ std::optional<SharedPreferencesForWebProcess> RemoteGraphicsContext::sharedPrefe
     return m_renderingBackend->sharedPreferencesForWebProcess();
 }
 
+Ref<RemoteImageBufferGraphicsContext> RemoteImageBufferGraphicsContext::create(WebCore::ImageBuffer& imageBuffer, RemoteGraphicsContextIdentifier identifier, RemoteRenderingBackend& renderingBackend)
```
I think you should add `RemoteImageBufferGraphicsContext.cpp`

smfr commented on Source/WebKit/GPUProcess/graphics/RemoteGraphicsContext.cpp:
```diff
@@ -767,6 +749,58 @@ std::optional<SharedPreferencesForWebProcess> RemoteGraphicsContext::sharedPrefe
+    m_renderingBackend->streamConnection().stopReceivingMessages(Messages::RemoteGraphicsContext::messageReceiverName(), m_identifier.toUInt64());
+}
+
+Ref<RemoteDisplayListRecorder> RemoteDisplayListRecorder::create(RemoteDisplayListRecorderIdentifier identifier, RemoteRenderingBackend& renderingBackend)
```
And `RemoteDisplayListRecorder .cpp`

smfr commented on Source/WebKit/GPUProcess/graphics/RemoteGraphicsContext.h:
```diff
@@ -198,6 +199,41 @@ class RemoteGraphicsContext : public IPC::StreamMessageReceiver, public CanMakeW
+
+// RemoteGraphicsContext playing back the IPC GraphicsContext drawing commands to a GraphicsContext of an ImageBuffer.
+// Used to create renderings to the ImageBuffer.
+class RemoteImageBufferGraphicsContext final : public RemoteGraphicsContext {
```
This should have its own header.

smfr commented on Source/WebKit/GPUProcess/graphics/RemoteGraphicsContext.h:
```diff
@@ -198,6 +199,41 @@ class RemoteGraphicsContext : public IPC::StreamMessageReceiver, public CanMakeW
+
+// RemoteGraphicsContext playing back the IPC GraphicsContext drawing commands to a DisplayList::Recorder.
+// Used to create DisplayList instances.
+class RemoteDisplayListRecorder final : public RemoteGraphicsContext {
```
Ditto.

smfr commented on Source/WebKit/GPUProcess/graphics/RemoteImageBufferGraphicsContext.cpp:
```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2021-2024 Apple Inc. All rights reserved.
```
2025

smfr commented on Source/WebKit/GPUProcess/graphics/RemoteImageBufferGraphicsContext.h:
```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2021-2024 Apple Inc. All rights reserved.
```
2025

kkinnunen-apple committed:
Implement GPU process side display lists
https://bugs.webkit.org/show_bug.cgi?id=297727
rdar://158865006

Reviewed by Simon Fraser.

Cache DisplayList data in GPUP and refer to them via
RemoteDisplayListIdentifier. Caching is done by constructing a
RemoteDisplayListRecorder IPC receiver that forwards the draw calls
to DisplayList::RecorderImpl to construct the DisplayList::DisplayList
instance.

Remove DecomposedGlyphs, it was overlapping and overly narrow
functionality.

Before, for a specific glyph run GlyphDisplayListCache would
drawGlyphs() to a DisplayList::RecorderImpl in DecomposeAndRetain mode.
This would decompose the single DrawGlyph call into Translate, Scale,
ConcatenateCTM, DrawDecomposedGlyphs, ... operations.
DrawDecomposedGlyphs would contain the DecomposedGlyphs structure, which
would be cached to GPUP upon first playback of the display list. Further
playbacks would again issue all commands of the display list, but reuse
the cached DecomposedGlyphs data.

After, we cache the whole display list. This way drawing the display
list is just one command.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/graphics/BifurcatedGraphicsContext.cpp:
(WebCore::BifurcatedGraphicsContext::drawDecomposedGlyphs): Deleted.
* Source/WebCore/platform/graphics/BifurcatedGraphicsContext.h:
* Source/WebCore/platform/graphics/DecomposedGlyphs.cpp: Removed.
* Source/WebCore/platform/graphics/DecomposedGlyphs.h: Removed.
* Source/WebCore/platform/graphics/FontCascade.cpp:
(WebCore::FontCascade::displayListForTextRun const):
* Source/WebCore/platform/graphics/GraphicsContext.cpp:
(WebCore::GraphicsContext::drawGlyphsImmediate):
(WebCore::GraphicsContext::drawDecomposedGlyphs): Deleted.
* Source/WebCore/platform/graphics/GraphicsContext.h:
* Source/WebCore/platform/graphics/NullGraphicsContext.h:
* Source/WebCore/platform/graphics/RenderingResource.h:
(WebCore::RenderingResource::isGradient const):
(WebCore::RenderingResource::isDecomposedGlyphs const): Deleted.
* Source/WebCore/platform/graphics/cairo/CairoOperationRecorder.cpp:
(WebCore::Cairo::OperationRecorder::drawDecomposedGlyphs): Deleted.
* Source/WebCore/platform/graphics/cairo/CairoOperationRecorder.h:
* Source/WebCore/platform/graphics/cairo/GraphicsContextCairo.cpp:
* Source/WebCore/platform/graphics/coretext/DrawGlyphsRecorder.cpp:
(WebCore::DrawGlyphsRecorder::DrawGlyphsRecorder):
(WebCore::DrawGlyphsRecorder::recordDrawGlyphs):
* Source/WebCore/platform/graphics/coretext/DrawGlyphsRecorder.h:
* Source/WebCore/platform/graphics/displaylists/DisplayList.cpp:
(WebCore::DisplayList::DisplayList::~DisplayList):
* Source/WebCore/platform/graphics/displaylists/DisplayList.h:
* Source/WebCore/platform/graphics/displaylists/DisplayListItem.h:
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.cpp:
(WebCore::DisplayList::DrawDecomposedGlyphs::apply const): Deleted.
(WebCore::DisplayList::DrawDecomposedGlyphs::dump const): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.h:
(WebCore::DisplayList::DrawDecomposedGlyphs::DrawDecomposedGlyphs): Deleted.
(WebCore::DisplayList::DrawDecomposedGlyphs::font const): Deleted.
(WebCore::DisplayList::DrawDecomposedGlyphs::decomposedGlyphs const): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorder.cpp:
(WebCore::DisplayList::Recorder::appendDisplayList):
(WebCore::DisplayList::Recorder::decomposeDrawGlyphsIfNeeded):
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorder.h:
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorderImpl.cpp:
(WebCore::DisplayList::RecorderImpl::drawDecomposedGlyphs): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorderImpl.h:
* Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp:
* Source/WebCore/rendering/GlyphDisplayListCache.cpp:
(WebCore::GlyphDisplayListCache::canShareDisplayList):
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
(WebCore::RenderThemeCocoa::paintAttachmentText):
* Source/WebKit/GPUProcess/graphics/RemoteGraphicsContext.cpp:
(WebKit::RemoteGraphicsContext::RemoteGraphicsContext):
(WebKit::RemoteGraphicsContext::controlFactory):
(WebKit::RemoteGraphicsContext::drawDisplayList):
(WebKit::RemoteGraphicsContext::drawControlPart):
(WebKit::RemoteImageBufferGraphicsContext::create):
(WebKit::RemoteImageBufferGraphicsContext::RemoteImageBufferGraphicsContext):
(WebKit::RemoteImageBufferGraphicsContext::startListeningForIPC):
(WebKit::RemoteImageBufferGraphicsContext::stopListeningForIPC):
(WebKit::RemoteDisplayListRecorder::create):
(WebKit::RemoteDisplayListRecorder::RemoteDisplayListRecorder):
(WebKit::RemoteDisplayListRecorder::startListeningForIPC):
(WebKit::RemoteDisplayListRecorder::stopListeningForIPC):
(WebKit::RemoteGraphicsContext::create): Deleted.
(WebKit::RemoteGraphicsContext::startListeningForIPC): Deleted.
(WebKit::RemoteGraphicsContext::stopListeningForIPC): Deleted.
(WebKit::RemoteGraphicsContext::drawDecomposedGlyphs): Deleted.
* Source/WebKit/GPUProcess/graphics/RemoteGraphicsContext.h:
(WebKit::RemoteGraphicsContext::context):
* Source/WebKit/GPUProcess/graphics/RemoteGraphicsContext.messages.in:
* Source/WebKit/GPUProcess/graphics/RemoteImageBuffer.cpp:
(WebKit::RemoteImageBuffer::RemoteImageBuffer):
* Source/WebKit/GPUProcess/graphics/RemoteImageBuffer.h:
* Source/WebKit/GPUProcess/graphics/RemoteImageBufferSet.cpp:
(WebKit::RemoteImageBufferSet::ensureBufferForDisplay):
* Source/WebKit/GPUProcess/graphics/RemoteImageBufferSet.h:
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.cpp:
(WebKit::RemoteRenderingBackend::createDisplayListRecorder):
(WebKit::RemoteRenderingBackend::sinkDisplayListRecorderIntoDisplayList):
(WebKit::RemoteRenderingBackend::releaseDisplayList):
(WebKit::RemoteRenderingBackend::cacheDecomposedGlyphs): Deleted.
(WebKit::RemoteRenderingBackend::releaseDecomposedGlyphs): Deleted.
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.h:
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.messages.in:
* Source/WebKit/GPUProcess/graphics/RemoteResourceCache.cpp:
(WebKit::RemoteResourceCache::cacheDisplayList):
(WebKit::RemoteResourceCache::cachedDisplayList const):
(WebKit::RemoteResourceCache::releaseDisplayList):
(WebKit::RemoteResourceCache::releaseMemory):
(WebKit::RemoteResourceCache::cacheDecomposedGlyphs): Deleted.
(WebKit::RemoteResourceCache::releaseDecomposedGlyphs): Deleted.
(WebKit::RemoteResourceCache::cachedDecomposedGlyphs const): Deleted.
* Source/WebKit/GPUProcess/graphics/RemoteResourceCache.h:
* Source/WebKit/Scripts/webkit/messages.py:
(atomic_object_identifier):
(serialized_identifiers):
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/graphics/RemoteDisplayListIdentifier.h: Copied from Source/WebKit/WebProcess/GPU/graphics/RemoteGraphicsContextIdentifier.h.
* Source/WebKit/WebProcess/GPU/graphics/RemoteGraphicsContextIdentifier.h:
* Source/WebKit/WebProcess/GPU/graphics/RemoteGraphicsContextProxy.cpp:
(WebKit::RemoteGraphicsContextProxy::RemoteGraphicsContextProxy):
(WebKit::m_renderingBackend):
(WebKit::RemoteGraphicsContextProxy::drawDisplayList):
(WebKit::RemoteGraphicsContextProxy::recordResourceUse):
(WebKit::RemoteGraphicsContextProxy::drawDecomposedGlyphs): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/RemoteGraphicsContextProxy.h:
* Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.cpp:
(WebKit::RemoteRenderingBackendProxy::cacheDisplayList):
(WebKit::RemoteRenderingBackendProxy::releaseDisplayList):
(WebKit::RemoteRenderingBackendProxy::cacheDecomposedGlyphs): Deleted.
(WebKit::RemoteRenderingBackendProxy::releaseDecomposedGlyphs): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.h:
* Source/WebKit/WebProcess/GPU/graphics/RemoteResourceCacheProxy.cpp:
(WebKit::RemoteResourceCacheProxy::recordDisplayListUse):
(WebKit::RemoteResourceCacheProxy::willDestroyDisplayList):
(WebKit::RemoteResourceCacheProxy::releaseMemory):
(WebKit::RemoteResourceCacheProxy::recordDecomposedGlyphsUse): Deleted.
(WebKit::RemoteResourceCacheProxy::willDestroyDecomposedGlyphs): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/RemoteResourceCacheProxy.h:

Canonical link: https://commits.webkit.org/299747@main
